sdk/dapp-kit/src/hooks/useSuiClientQuery.ts

```diff
@@ -29,20 +30,23 @@ export type SuiRpcMethods = {
 		: never;
 };
 
-export type UseSuiClientQueryOptions<T extends keyof SuiRpcMethods> = Omit<
-	UseQueryOptions<SuiRpcMethods[T]['result'], Error, SuiRpcMethods[T]['result'], unknown[]>,
-	'queryFn'
+export type UseSuiClientQueryOptions<T extends keyof SuiRpcMethods, TData> = PartialBy<
+	Omit<UseQueryOptions<SuiRpcMethods[T]['result'], Error, TData, unknown[]>, 'queryFn'>,
+	'queryKey'
 >;
 
-export function useSuiClientQuery<T extends keyof SuiRpcMethods>(
+export function useSuiClientQuery<
```

Comment:
hey @hayes-mysten! would you mind sanity-checking the typing for `TData` I added here? 

Basically, we want to allow folks to specify a custom return type for the `data` field to allow transformations via `select`. By default though, we'll just use `SuiRpcMethods[T]['result']`

Reply:
Looks good to me!